apps/hash-api/src/generate-ontology-type-ids.ts

```diff
@@ -0,0 +1,285 @@
+
+const convertTitleToCamelCase = (title: string) =>
+  title
+    .split(" ")
+    .map((word, index) =>
+      // If it's the first word, convert it to lowercase
+      // Otherwise, capitalize the first letter and then add the rest of the word in lowercase
+      index === 0
+        ? word.toLowerCase()
+        : word.charAt(0).toUpperCase() + word.slice(1).toLowerCase(),
+    )
+    // Join all the processed words to get the camelCase result
+    .join("");
+
+const generateOntologyIds = async () => {
+  const logger = new Logger({
+    mode: "dev",
+    level: "debug",
+    serviceName: "generate-ontology-ids",
+  });
+
+  const graphApiHost = getRequiredEnv("HASH_GRAPH_API_HOST");
+  const graphApiPort = parseInt(getRequiredEnv("HASH_GRAPH_API_PORT"), 10);
+
+  const graphApi = createGraphClient(logger, {
+    host: graphApiHost,
+    port: graphApiPort,
+  });
+
+  const graphContext = { graphApi };
+
+  const hashOrg = await getOrgByShortname(
+    graphContext,
+    { actorId: publicUserAccountId },
+    { shortname: "hash" },
+  );
+
+  if (!hashOrg) {
+    throw new Error("HASH org not found");
+  }
+
+  const authentication = { actorId: publicUserAccountId };
+
+  const [
+    allSystemEntityTypes,
+    systemPropertyTypes,
+    blockProtocolEntityTypes,
+    blockProtocolPropertyTypes,
+    blockProtocolDataTypes,
+  ] = await Promise.all([
+    getEntityTypes(graphContext, authentication, {
+      query: {
+        filter: {
+          all: [
+            {
+              equal: [{ path: ["version"] }, { parameter: "latest" }],
+            },
+            {
+              equal: [
+                {
+                  path: ["ownedById"],
+                },
+                { parameter: hashOrg.accountGroupId },
+              ],
+            },
+          ],
+        },
+        graphResolveDepths: zeroedGraphResolveDepths,
+        temporalAxes: currentTimeInstantTemporalAxes,
+      },
+    }).then((subgraph) => getRoots(subgraph)),
+    getPropertyTypes(graphContext, authentication, {
+      query: {
+        filter: {
+          all: [
+            {
+              equal: [{ path: ["version"] }, { parameter: "latest" }],
+            },
+            {
+              equal: [
+                { path: ["ownedById"] },
+                { parameter: hashOrg.accountGroupId },
+              ],
+            },
+          ],
+        },
+        graphResolveDepths: zeroedGraphResolveDepths,
+        temporalAxes: currentTimeInstantTemporalAxes,
+      },
+    }).then((subgraph) => getRoots(subgraph)),
+    getEntityTypes(graphContext, authentication, {
+      query: {
+        filter: {
+          all: [
+            {
+              equal: [{ path: ["version"] }, { parameter: "latest" }],
+            },
+            {
+              startsWith: [
+                { path: ["versionedUrl"] },
+                { parameter: "https://blockprotocol.org" },
+              ],
+            },
```

Comment:
@CiaranMn because these are external types, my understanding is that they're not in an org which I can query the `ownedById` via. Let me know if there's a better way for obtaining these types